process/process_linux.go

```diff
@@ -49,6 +90,110 @@ type MemoryMapsStat struct {
 	Swap         uint64 `json:"swap"`
 }
 
+func (m *MemoryMapsStat) updateAttribute(fields []string) error {
+	if len(fields) < 2 {
+		return fmt.Errorf("update attribute not enough fields")
+	}
+	var v *uint64
+	switch fields[0] {
+	case "Size:":
+		v = &m.Size
+	case "Rss:":
+		v = &m.Rss
+	case "Pss:":
+		v = &m.Pss
+	case "Shared_Clean:":
+		v = &m.SharedClean
+	case "Shared_Dirty:":
+		v = &m.SharedDirty
+	case "Private_Clean:":
+		v = &m.PrivateClean
+	case "Private_Dirty:":
+		v = &m.PrivateDirty
+	case "Referenced:":
+		v = &m.Referenced
+	case "Anonymous:":
+		v = &m.Anonymous
+	case "Swap:":
+		v = &m.Swap
+	default:
+		/* ignore other fields */
+		return nil
+	}
+
+	t, err := strconv.ParseUint(fields[1], 10, 64)
+	if err != nil {
+		return err
+	}
+
+	*v = t
+	return nil
+}
+
+/*
+   Parse a memory section from /proc/pid/maps or header section line of /proc/pid/smaps
+
+ StartAddr        EndAddr    Permission Offset Device Inode              Path
+ffffffffff600000-ffffffffff601000 --xp 00000000 00:00 0                  [vsyscall]
+*/
+func (ms *MemoryMapsStat) parseHeader(fields []string) error {
+	if len(fields) < 5 {
+		return fmt.Errorf("can't parse section, not enough fields '%v'", fields)
+	}
+
+	/* Address Range */
+	addrs := strings.Split(fields[0], "-")
```

Comment:
Question: is `parseHeader` gonna be called a lot? If so, doing `Split` will allocate memory and in the past this would cause resource usage increase. A more efficient approach would be manually iterating the string and test the substring, but if it's only used sporadically then it's not worth the effort.

Reply:
It's kind of used around 23 times per pid, I switched to strings.IndexRune() and split the slice manually